src/shared/utils/api.js

```diff
@@ -0,0 +1,87 @@
+/* eslint-disable no-console */
+import { userHistoryApiUrl, userHistorySchema } from '../data/keys';
+
+const historyApiUrl = `${userHistoryApiUrl}/v1`;
+
+// TODO: two ways to handle error when fetching, another way is move these apis to redux toolkit
+const fetchData = async (url, options = {}) => {
```

Comment:
should this be a thunk like the other api calls?

Reply:
These don't have to be thunks since they are not updating values in redux. But this might be possible to handle errors(using redux to control alert component).